lorien/UI/Dialogs/AddKeyDialog.gd

```diff
@@ -0,0 +1,62 @@
+extends WindowDialog
+
+# -------------------------------------------------------------------------------------------------
+export var action_name: String
+export var readable_action_name: String
+
+# -------------------------------------------------------------------------------------------------
+var _pending_bind_event = null
+
+# -------------------------------------------------------------------------------------------------
+func _process(delta):
+	$VBoxContainer/EventText.text = "Action: %s" % readable_action_name
+
+# -------------------------------------------------------------------------------------------------
+func _action_for_event(event: InputEvent):
+	for action in Utils.bindable_actions():
+		if InputMap.action_has_event(action, event):
+			return action
+	return null
+
+# -------------------------------------------------------------------------------------------------
+func _input(event: InputEvent) -> void:
+	if ! visible or $ConfirmRebind.visible:
+		return
+	
+	if event is InputEventKey && event.is_pressed():
+		get_tree().set_input_as_handled()
```

Comment:
This is actually meant to be `_input`. This is what all the earlier work was for. `set_input_as_handled` makes sure that the function does not trigger any other actions and allows the dialog to process all strokes to assemble a InputEventKey.